src/server/pachyderm_test.go

```diff
@@ -2927,6 +2928,200 @@ func TestDatumStatusRestart(t *testing.T) {
 	require.Equal(t, 1, len(commitInfos))
 }
 
+// TestSystemResourceRequest doesn't create any jobs or pipelines, it
+// just makes sure that when pachyderm is deployed, we give rethinkdb, pachd,
+// and etcd default resource requests. This prevents them from overloading
+// nodes and getting evicted, which can slow down or break a cluster.
+func TestSystemResourceRequests(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	t.Parallel()
+	kubeClient := getKubeClient(t)
+
+	// Get Pod info for 'app' from k8s
+	var c api.Container
+	for _, app := range []string{"pachd", "etcd"} {
+		b := backoff.NewExponentialBackOff()
+		b.MaxElapsedTime = 10 * time.Second
+		err := backoff.Retry(func() error {
+			podList, err := kubeClient.Pods(api.NamespaceDefault).List(api.ListOptions{
+				LabelSelector: labels.SelectorFromSet(
+					map[string]string{"app": app, "suite": "pachyderm"}),
+			})
+			if err != nil {
+				return err
+			}
+			if len(podList.Items) < 1 {
+				return fmt.Errorf("could not find pod for %s", app) // retry
+			}
+			c = podList.Items[0].Spec.Containers[0]
+			return nil
+		}, b)
+		require.NoError(t, err)
+
+		// Make sure the pod's container has resource requests
+		_, ok := c.Resources.Requests[api.ResourceCPU]
```

Comment:
Would it make sense to check that this value is equal to the defaults?  Are we sure that this check fails if resource requests were not specified?  Cuz I can see a world where k8s automatically sets some default resource requests and this check succeeds regardless.